### PR TITLE
IOP (IBM Open Platform) images

### PR DESCRIPTION
**Pull request by ArturGajowy**
Supersedes #10 (contains its changes and fixups for them). @kokosing please have another look, at least at the test infrastructure.

### CODE REVIEW

File: depend.sh [hunk 1]
```diff
@@ -81,6 +81,8 @@ if contains "$parent_image" "$known_images"; then
 	# The parent image is built from the repository
 	if ! "$list_external"; then
 		echo "$target_image: $parent_image"
+		echo "$target_image.dependants: $target_image"
+		echo "$parent_image.dependants: $target_image.dependants"
```
Comment (kokosing) [inline depend.sh:85]: this seems to belong to different commit

File: pull_latest_images.sh [hunk 2]
```diff
@@ -10,7 +10,5 @@ for i in teradatalabs/*; do
 done
 
 for PID in $PIDS; do
-  wait $PID || let "RESULT=$?"
-done
-
-exit $RESULT
\ No newline at end of file
+  wait $PID
+done
```
Comment (kokosing) [inline pull_latest_images.sh]: no new line at the end

File: teradatalabs/iop-base/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM teradatalabs/centos6-java8-oracle
+MAINTAINER Teradata Docker Team <docker@teradata.com>
+
+RUN wget -nv https://ibm-open-platform.ibm.com/repos/IOP/rhel/6/x86_64/4.2.x/GA/IOP.repo -P /etc/yum.repos.d
+
+RUN yum install -y hive_4_2_0_0
```
Comment (kokosing) [inline teradatalabs/iop-base/Dockerfile]: why this two commands are separated?

File: teradatalabs/hdp2.5-hive/files/setup.sh [hunk 4]
```diff
@@ -1,6 +1,4 @@
-#!/bin/bash
-
-set -ex
```
Comment (kokosing) [inline teradatalabs/hdp2.5-hive/files/setup.sh:3]: this seems to be belong to different commit

File: teradatalabs/iop-hive/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+# INSTALL HADOOP AND HIVE
+RUN \
+  yum install -y \
+    zookeeper_4_2_0_0-server \
+  && ln -s /usr/iop/4.2.0.0/zookeeper/bin/zookeeper-server /usr/bin/zookeeper-server \
+  && yum -y clean all && rm -rf /tmp/* /var/tmp/*
+
+# INSTALL MYSQL (Hive Metastore)
+RUN yum install -y mysql-server mysql-connector-java \
```
Comment (kokosing) [inline teradatalabs/iop-hive/Dockerfile:26]: why these installers are separated?

File: teradatalabs/iop-hive/files/bootstrap.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Script executed by supervisord as bootstrap
+####
+
+# It manager the hadoop processes startup sequence via
+# the systemctl command. This is the only script launched
+# at startup by supervisord.
+
+# 1 Zookeeper
```
Comment (kokosing) [inline teradatalabs/iop-hive/files/bootstrap.sh]: remove dummy comments

**Review by kokosing — APPROVED** [review]
% some small comments

File: teradatalabs/iop-hive/Dockerfile (hunk 5 above)
Comment (ArturGajowy) [reply teradatalabs/iop-hive/Dockerfile:26]: Mainly for consistency with other `*-hive` files and because sorting it out in all of them (and in all other *-master/slave images) is beyond the amount of work I want to spend on this... :)
